### PR TITLE
External members and institutions crud

### PR DESCRIPTION
**Pull request by renangabriel27**

### CODE REVIEW

File: app/controllers/responsible/external_members_controller.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+  add_breadcrumb I18n.t('breadcrumbs.external_members.new'),
+                 :new_responsible_external_member_path,
+                 only: [:new]
+
+  add_breadcrumb I18n.t('breadcrumbs.external_members.edit'),
+                 :edit_responsible_external_member_path,
+                 only: [:edit]
+
+  def index
```
Comment (dmarczal) [inline app/controllers/responsible/external_members_controller.rb]: Use like this:
```
external_members = ExternalMember.page(params[:page]).search(params[:term])
```

File: app/controllers/responsible/external_members_controller.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  def new
+    @external_member = ExternalMember.new
+  end
+
+  def edit; end
+
+  def create
+    @external_member = ExternalMember.new(external_member_params)
+    @external_member.skip_password_validation = true
```
Comment (dmarczal) [inline app/controllers/responsible/external_members_controller.rb]: External member needs a password.

File: app/controllers/responsible/external_members_controller.rb (hunk 1 above)
Comment (dmarczal) [reply app/controllers/responsible/external_members_controller.rb]: Review others controllers

File: app/models/external_member.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  mount_uploader :profile_image, ProfileImageUploader
+
+  def self.human_genders
+    hash = {}
+    genders.each_key { |key| hash[I18n.t("enums.genders.#{key}")] = key }
+    hash
+  end
+
+  def self.filter_by_company_responsible
```
Comment (dmarczal) [inline app/models/external_member.rb]: The will not be necessary.

**Comment by renangabriel27** [comment]
> 1. You should add a new professor_title called "Graduado", "Ensino médio completo", because this can be the title for an external member. Notice that the name professor_title doesn't make sense for the external member, because it may not be a professor. Maybe yes, maybe not. If this table is used for both register rename it to "scholarity (escolaridade)"
> 2. When I have no registers for institutions this error occurs:
> 
> NoMethodError at /responsible/institutions/new
> undefined method `id' for nil:NilClass
> 
> 1. This types are not necessary:
> 
> external_member_types = [
> 'Responsável de empresa',
> 'Coorientador externo',
> 'Membro de banca'
> ]
> 
> A member external can be any of these three options. He is going to be according to the relationship. For example for evaluator (membro de banca), the responsible teacher can select a professor or an external member. This will solve the problem 2, that filter is not necessary. You can select any external member as instution responsable.
> 
> 1. How about the password of the external member? Did you think about that? Maybe you can send an email notice the external member about the register, including the password and information to access the system.
> 2. How an external will get a new password if he forgot his password?
> 
> * Will the responsible teacher change his password?
> * Link to reset the password? I guess this option is better.
> * He can change his password on his perfil too.
> 
> 1. For external institutions put an optional field called working area, as a markdown.
> 2. Rember the difference about let and let!

fixed

**Pull request merged**